lib/ts/recipe/webauthn/api/implementation.ts

```diff
@@ -0,0 +1,1095 @@
+                  session: SessionContainerInterface;
+                  user: User;
+              }
+            | { status: "WRONG_CREDENTIALS_ERROR" }
+            | {
+                  status: "SIGN_IN_NOT_ALLOWED";
+                  reason: string;
+              }
+            | GeneralErrorResponse
+        > {
+            const errorCodeMap = {
+                SIGN_IN_NOT_ALLOWED:
+                    "Cannot sign in due to security reasons. Please try recovering your account, use a different login method or contact support. (ERR_CODE_008)",
+                LINKING_TO_SESSION_USER_FAILED: {
+                    EMAIL_VERIFICATION_REQUIRED:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_009)",
+                    RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_010)",
+                    ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_011)",
+                    SESSION_USER_ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR:
+                        "Cannot sign in / up due to security reasons. Please contact support. (ERR_CODE_012)",
+                },
+            };
+
+            const recipeId = "webauthn";
+
+            // do the verification before in order to retrieve the user email
+            const verifyCredentialsResponse = await options.recipeImplementation.verifyCredentials({
+                credential,
+                webauthnGeneratedOptionsId,
+                tenantId,
+                userContext,
+            });
+            const checkCredentialsOnTenant = async () => {
+                return verifyCredentialsResponse.status === "OK";
+            };
+
+            // doing it like this because the email is only available after verifyCredentials is called
+            let email: string;
+            if (verifyCredentialsResponse.status == "OK") {
+                const loginMethod = verifyCredentialsResponse.user.loginMethods.find((lm) => lm.recipeId === recipeId);
+                // there should be a webauthn login method  and an email when trying to sign in using webauthn
+                if (!loginMethod || !loginMethod.email) {
+                    return AuthUtils.getErrorStatusResponseWithReason(
+                        verifyCredentialsResponse,
+                        errorCodeMap,
+                        "SIGN_IN_NOT_ALLOWED"
+                    );
+                }
+                email = loginMethod?.email;
+            } else {
+                return {
+                    status: "WRONG_CREDENTIALS_ERROR",
+                };
+            }
+
+            const authenticatingUser = await AuthUtils.getAuthenticatingUserAndAddToCurrentTenantIfRequired({
```

Comment:
this function also needs to be updated to take the new account info prop into account (plus I think you'll need to check all places where we use the account info)

Reply:
maybe we could have a chat about this.

Reply:
same as previous comments. also need to check method to see if ti works with webauthn